libvast/src/pipeline.cpp

```diff
@@ -120,42 +119,48 @@ caf::expected<std::vector<table_slice>> pipeline::finish() {
     VAST_ASSERT_CHEAP(!import_timestamps_.empty());
     auto max_import_timestamp
       = *std::max_element(import_timestamps_.begin(), import_timestamps_.end());
-    for (auto& [schema, batch] : *finished) {
-      auto& slice = result.emplace_back(batch);
-      slice.import_time(max_import_timestamp);
+    for (auto& slice : *finished) {
+      if (slice.rows() == 0)
+        continue;
+      if (slice.import_time() == time{})
+        slice.import_time(max_import_timestamp);
+      result.push_back(std::move(slice));
     }
   } else {
-    for (size_t i = 0; auto& [schema, batch] : *finished) {
-      auto& slice = result.emplace_back(batch);
-      slice.import_time(import_timestamps_[i++]);
+    for (size_t i = 0; auto& slice : *finished) {
+      if (slice.rows() == 0)
+        continue;
+      if (slice.import_time() == time{})
+        slice.import_time(import_timestamps_[i++]);
+      result.push_back(std::move(slice));
     }
   }
   return result;
 }
 
-caf::error pipeline::add_batch(vast::type schema,
-                               std::shared_ptr<arrow::RecordBatch> batch) {
+caf::error pipeline::add_slice(table_slice slice) {
   VAST_TRACE("add arrow data to transform {}", name_);
-  to_transform_.emplace_back(std::move(schema), std::move(batch));
+  to_transform_.emplace_back(std::move(slice));
   return caf::none;
 }
 
-caf::error pipeline::process_queue(const std::unique_ptr<pipeline_operator>& op,
-                                   std::vector<pipeline_batch>& result,
-                                   bool check_schema) {
+caf::error
+pipeline::process_queue(const std::unique_ptr<pipeline_operator>& op,
```

Comment:
Not related to the changes. Feel free to drop. 
Unique_ptr part seems not used here. I think the parameter should just be pipeline_operator&